### ISSUE TITLE
Is this HA addon Cloud Polling or local Polling?

### ISSUE THREAD

**Issue by ceaswaran** [peterbuga/HASS-sonoff-ewelink#24]
Is this HA addon Cloud Polling or local Polling?

**Comment by peterbuga** [comment]
@ceaswaran cloud polling -> it acts **exactly** like ewelink app. although i fiddled with the idea of trying the "local scan" that it's present in the app, it's buggy and half-working most of the time so i dropped it.

more details here https://community.home-assistant.io/t/sonoff-ewelink-switch-component-for-original-firmware/81369

**Comment by ceaswaran** [comment]
@peterbuga  Thanks for the clarification. I appreciate.

I/ any user switched from ewelink app to custom firmware (Tasmota) for the 2 main reason

1. Cloud Polling as we don't want our data on someone's cloud
2. ewelink alway buggy/most of the time offline


Just curious to know what the advantage of using this custom component considering above 2 points

Please clarify

**Comment by peterbuga** [comment]
@ceaswaran I started this project just as a way to learn how the awesomeness of Home Assistant works and maybe add more features in the future (this is the first thing that I came up with as I saw 
 eWeLink was sort of ignored). 
your points are right on spot I couldn't agree more [1] 👍 (as I've kept mentioning in the thread above) but in the same time, as a quick example, I'm an Alexa user and just like Google Assistant the "HUE hack" could easily become unavailable or buggy over night. 

you or any other person for that matter can choose to use the original firmware or make the jump to custom one, you might have some IT-related background knowledge or simply willing to fiddle with stuff but I know a lot of people who simply like to keep the things as they are (personally I was too lazy to buy a FTDI device to flash them) and this component gives them this option. 

[2] if you take the time to decompile and take a quick look at the eWeLink source code (it's in javascript, that's what makes it big pile of 💩 ) you'll see the huge spaghetti code that it's there trying to make countless versions of devices to work, but the underlying API requests are actually quite basic and this is the part that I'm using. in other words this component _behaves_ like the app but it's _not like_ the app.

I haven't dug so much into Tasmota but I saw reports that they need to be recalibrated on models that report the power etc => more things to fiddle more time to waste, with this you just add 3 config lines and you're done 😉 

ps: there are A LOT of other official components in HA that use the same paradigm: they use the official cloud API in other to pull the data
ps2: I'm personally migrating to Shelly devices, they're far better than Sonoffs, but I'll continue to support this project as much as I can

I hope this clears some waters 😄
